### DISCUSSION TITLE
Is PhantomCamera3D designed to support both Follow and LookAt at the same time?

### DISCUSSION THREAD

**Discussion by ZenithStar** [ramokz/phantom-camera#196] [Question / Help Wanted] [answered]
Just to confirm, since the documentation doesn't give any examples of this, but the code flows in a way that this appears to work. FollowMode dictates how the position is determined while LookAtMode dictates how the orientation is handled.

**Comment by ramokz** [comment, marked as answer]
Short answer, yes — it's meant to support both.

What hasn't been done yet is proper testing and logic intermixing between the `Follow` and `Look At` logic. Hence, why the `Look At` logic is a bit sparse by comparison.

**Reply by ZenithStar** [reply]
Sounds good. Here's a rough idea of the kind of stuff I'm building right now.
[2024-01-25 12-10-57.webm](https://github.com/ramokz/phantom-camera/assets/12200619/77ac87af-135d-4978-bfef-b5b4cf1d96dd)

This is a single camera in third person follow mode and group look-at mode. It's kind of jarring right now without any rotational dampening, but should look much smoother once that's in. I can even look into writing that, but probably after the 0.7 stuff is merged in.

**Reply by ramokz** [reply]
Can definitely see what you mean by it missing that interpolation damping when switching between targets.
Cool demo scene btw :)

And yes, it would be a lot easier to implement after 0.7 as pretty much all the properties and functions are being refactored (#169), so would need to be reworked at that point in time anyway. Not to mention be better tested with the updated damping logic planned for it. You're absolutely more than welcome to have a go at it then, or even before, once #92 is done.
